File: Lesson_11_Differences_Between_MVC_MVP_MVVM/part_3_task_manager.py
class TaskModel:
    def __init__(self):
        self.tasks = []

    def add_task(self, task_name):
        self.tasks.append(task_name)

    def get_tasks(self):
        return self.tasks


class TaskView:
    def display_tasks(self, tasks):
        print("Current Tasks:")
        for task in tasks:
            print(f" - {task}")

    def get_task_input(self):
        return input("Enter a new task: ")


class TaskPresenter:
    def __init__(self, view, model):
        self.view = view
        self.model = model

    def add_task(self):
        task_input = self.view.get_task_input()
        self.model.add_task(task_input)
        self.update_view()

    def update_view(self):
        tasks = self.model.get_tasks()
        self.view.display_tasks(tasks)


# Usage
model = TaskModel()
view = TaskView()
presenter = TaskPresenter(view, model)

presenter.add_task()

File: Lesson_12_Summary_of_Modern_MV_Patterns/part_3_taskManager.js
class TaskModel {
  constructor() {
    this.tasks = [];
  }

  addTask(taskName) {
    this.tasks.push(taskName);
  }

  getTasks() {
    return this.tasks;
  }
}

class TaskViewModel {
  constructor(model) {
    this.model = model;
    this.tasks = [];
  }

  loadTasks() {
    this.tasks = this.model.getTasks();
  }

  addTask(taskName) {
    this.model.addTask(taskName);
    this.loadTasks();
  }
}

const model = new TaskModel();
const viewModel = new TaskViewModel(model);
viewModel.addTask("Learn MV Patterns");
console.log(viewModel.tasks);
